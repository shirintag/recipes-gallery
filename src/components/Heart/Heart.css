.heart {
  background-color: rgba(201, 201, 201);
  height: 16px;
  transform: rotate(-45deg);
  width: 16px;
  cursor: pointer;
}

.heart:before,
.heart:after {
  content: "";
  background-color: rgba(201, 201, 201);
  border-radius: 50%;
  height: 16px;
  position: absolute;
  width: 16px;
}

.heart:before {
  top: -8px;
  left: 0;
}

.heart.selected {
  background-color: red;
}

.heart.selected:before,
.heart.selected:after {
  background-color: red;
}

.heart.selected:before {
  background-color: red;
}
