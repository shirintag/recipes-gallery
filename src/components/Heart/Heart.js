import React, { useState } from "react";

import "./Heart.css";

const Heart = () => {
  const [selectedRecipe, selectRecipe] = useState(false);

  return (
    <div>
      <div
        className={selectedRecipe ? "heart selected" : "heart"}
        onClick={() => selectRecipe(!selectedRecipe)}
      />
    </div>
  );
};

export default Heart;
