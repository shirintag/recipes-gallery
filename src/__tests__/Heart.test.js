import React from "react";
import { render } from "@testing-library/react";
import Heart from "components/Heart/Heart";

describe("Heart", () => {
  it("should render", () => {
    const { container } = render(<Heart />);
    expect(container).toMatchSnapshot();
  });

  it("should change class on click", () => {
    let { container } = render(<Heart />);
    container = container.firstChild.firstChild;
    expect(container.className).toEqual("heart");

    container.click();
    expect(container.className).toEqual("heart selected");
  });
});
