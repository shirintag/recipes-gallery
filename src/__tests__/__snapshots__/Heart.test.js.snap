// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heart should render 1`] = `
<div>
  <div>
    <div
      class="heart"
    />
  </div>
</div>
`;
